src/test/unit/algorithms/SpatialPoolerTest.cpp

```diff
@@ -1487,85 +1501,80 @@ TEST(SpatialPoolerTest, testInitPermConnected) {
   }
 }
 
+
 TEST(SpatialPoolerTest, testInitPermNonConnected) {
   Real32 synPermConnected = 0.2f;
-  SpatialPooler sp({10}, {10}, 16u, 0.5f, true, -1, 10u, 0u, 0.01f, 0.1f,
+  SpatialPooler sp({10}, {10}, 16u, 0.5f, true, 0.5f, 0u, 0.01f, 0.1f,
                    synPermConnected);
-  EXPECT_TRUE(sp.getSynPermMax() == 1.0) << sp.getSynPermMax();
+  EXPECT_EQ(sp.getSynPermMax(), 1.0f);
+
   for (UInt i = 0; i < 100; i++) {
     Real permVal = sp.initPermNonConnected_();
-    ASSERT_GE(permVal, 0);
+    ASSERT_GE(permVal, 0.0f);
     ASSERT_LE(permVal, synPermConnected);
   }
 }
 
+
 TEST(SpatialPoolerTest, testinitMapColumn) {
   {
     // Test 1D.
-    SpatialPooler sp(
-        /*inputDimensions*/ {12},
-        /*columnDimensions*/ {4});
+    SpatialPooler sp;
+    setup(sp, /*inputDimensions*/ {12}, /*columnDimensions*/ {4});
 
     EXPECT_EQ(1ul, sp.initMapColumn_(0));
     EXPECT_EQ(4ul, sp.initMapColumn_(1));
     EXPECT_EQ(7ul, sp.initMapColumn_(2));
-    EXPECT_EQ(10ul, sp.initMapColumn_(3));
+    EXPECT_EQ(10ul,sp.initMapColumn_(3));
   }
 
   {
     // Test 1D with same dimensions of columns and inputs.
-    SpatialPooler sp(
-        /*inputDimensions*/ {4},
-        /*columnDimensions*/ {4});
+    SpatialPooler sp;
+    setup(sp, /*inputDimensions*/ {4}, /*columnDimensions*/ {4});
 
     EXPECT_EQ(0ul, sp.initMapColumn_(0));
     EXPECT_EQ(1ul, sp.initMapColumn_(1));
     EXPECT_EQ(2ul, sp.initMapColumn_(2));
     EXPECT_EQ(3ul, sp.initMapColumn_(3));
   }
 
-  {
-    // Test 1D with dimensions of length 1.
-    SpatialPooler sp(
-        /*inputDimensions*/ {1},
-        /*columnDimensions*/ {1});
```

Comment:
deleted this dummy test as it's already too artificial. And I even cannot pass sparsity=100% to setup() because we have internal SP check for max sparsity <= 50%.